crates/runtime/src/flight/actions.rs

```diff
@@ -0,0 +1,100 @@
+            "ClosePreparedStatement" => ActionType::ClosePreparedStatement,
+            _ => ActionType::Unknown,
+        }
+    }
+
+    fn as_str(&self) -> &'static str {
+        match self {
+            ActionType::CreatePreparedStatement => "CreatePreparedStatement",
+            ActionType::ClosePreparedStatement => "ClosePreparedStatement",
+            ActionType::Unknown => "Unknown",
+        }
+    }
+}
+
+impl Display for ActionType {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.as_str())
+    }
+}
+
+pub(crate) fn list() -> Response<<Service as FlightService>::ListActionsStream> {
+    tracing::trace!("list_actions");
+    let create_prepared_statement_action_type = FlightActionType {
+        r#type: ActionType::CreatePreparedStatement.to_string(),
+        description: "Creates a reusable prepared statement resource on the server.\n
+            Request Message: ActionCreatePreparedStatementRequest\n
+            Response Message: ActionCreatePreparedStatementResult"
+            .into(),
+    };
+    let close_prepared_statement_action_type = FlightActionType {
+        r#type: ActionType::ClosePreparedStatement.to_string(),
+        description: "Closes a reusable prepared statement resource on the server.\n
+            Request Message: ActionClosePreparedStatementRequest\n
+            Response Message: N/A"
+            .into(),
+    };
+    let actions: Vec<Result<FlightActionType, Status>> = vec![
+        Ok(create_prepared_statement_action_type),
+        Ok(close_prepared_statement_action_type),
+    ];
+
+    let output = futures::stream::iter(actions);
+    Response::new(Box::pin(output) as <Service as FlightService>::ListActionsStream)
+}
+
+pub(crate) async fn do_action(
+    flight_svc: &Service,
+    request: Request<Action>,
+) -> Result<Response<<Service as FlightService>::DoActionStream>, Status> {
+    let action_type = ActionType::from_str(request.get_ref().r#type.as_str());
+
+    match action_type {
+        ActionType::CreatePreparedStatement => {
+            tracing::trace!("do_action: CreatePreparedStatement");
+            let any = Any::decode(&*request.get_ref().body).map_err(to_tonic_err)?;
+
+            let cmd: sql::ActionCreatePreparedStatementRequest =
+                any.unpack().map_err(to_tonic_err)?.ok_or_else(|| {
+                    Status::invalid_argument(
+                        "Unable to unpack ActionCreatePreparedStatementRequest.",
+                    )
+                })?;
+            let stmt =
+                prepared_statement_query::do_action_create_prepared_statement(flight_svc, cmd)
+                    .await?;
+            let output = futures::stream::iter(vec![Ok(arrow_flight::Result {
+                body: stmt.as_any().encode_to_vec().into(),
+            })]);
+            Ok(Response::new(Box::pin(output)))
+        }
+        ActionType::ClosePreparedStatement => {
```

Comment:
This is essentially a noop because we don't instantiate any state for the `CreatePreparedStatement`?

Reply:
Yep